CMakeLists.txt

```diff
@@ -26,13 +38,14 @@ set(CMAKE_C_EXTENSIONS OFF)
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 
-option(BUILD_SHARED_LIBS "Build shared libraries" ON)
+option(BUILD_SHARED_LIBS "Build shared libraries." ON)
 option(SECP256K1_DISABLE_SHARED "Disable shared library. Overrides BUILD_SHARED_LIBS." OFF)
 if(SECP256K1_DISABLE_SHARED)
   set(BUILD_SHARED_LIBS OFF)
 endif()
 
-option(SECP256K1_INSTALL "Enable installation" ON)
+include(CMakeDependentOption)
+cmake_dependent_option(SECP256K1_INSTALL "Enable installation." ON "PROJECT_IS_TOP_LEVEL" OFF)
```

Comment:
Does it really make sense to forbid installation if used as a subproject? I think we could let the user decide here.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/secp256k1/pull/1284#issuecomment-1517661110).